kahypar/partition/multilevel.h

```diff
@@ -124,5 +126,45 @@ static inline void partition(Hypergraph& hypergraph,
 
   io::printLocalSearchResults(context, hypergraph);
 }
+
+static inline void partitionRepeatedOnInfeasible(Hypergraph& hypergraph,
+                                                 const Context& context,
+                                                 Context::PartitioningStats& stats,
```

Comment:
Is there a specific reason that the stats are passed as an additional non-const reference? The stats of the context object are mutable so they can be changed even if the context is passed in as const ref.

Reply:
Yes. To this method we pass not the original context but the context of the current bisection. I can use the stats of the current context, but then the information is lost after the current bisection. Thus the stats of the original context need to be passed additionally.

Of course a possible solution could be to share the stats between copies of the context (via shared_ptr?), but this would probably require changes at other locations, too.

Reply:
Okay, I see. In that case, your approach totally makes sense!